docs/conf.py

```diff
@@ -233,6 +237,36 @@
 
 """
 
+
+# ----------------------------------------------------------------------------------
+# Redirects
+# ----------------------------------------------------------------------------------
+
+def determine_api_redirects() -> dict[str, str]:
+    """Set up API redirects for functions that we moved to module pages.
+
+    Note that we have redirects in Cloudflare for methods moving to their class page. We
+    could not do this for functions because some functions still have dedicated
+    HTML pages, so we cannot use a generic rule.
```

Comment:
We're using Cloudflare for a lot: 

* all the Ecosystem redirects for what used to be `documentation/<project>`
* The Aer API docs that used to be under Qiskit's API docs. We had two migrations here
* As of yesterday, Qiskit methods now going to class pages
* several tutorials migrations
* Textbook/learning platform stuff

Point being, we will need to keep the Cloudflare redirects for qiskit.org in general and it's impossible for sphinx-reredirects to handle it all since it only helps with the /documentation subsite. So, I don't see much reason to move some of Cloudflare here.